dev/build-tool-support/src/main/scala/com/lightbend/lagom/dev/ManifestInfo.scala

```diff
@@ -0,0 +1,235 @@
+        None
+    }
+
+}
+
+object ManifestInfo {
+
+  private val ImplTitle = "Implementation-Title"
+  private val ImplVersion = "Implementation-Version"
+  private val ImplVendor = "Implementation-Vendor-Id"
+
+  private val BundleName = "Bundle-Name"
+  private val BundleVersion = "Bundle-Version"
+  private val BundleVendor = "Bundle-Vendor"
+
+  // TODO: reconsider this if we want to make it more generic
+  // instead we should collect the versions only when doing the check and ProjectFamily should define the vendor id
+  private val knownVendors = Set(
+    "com.typesafe.akka",
+    "com.lightbend.akka",
+    "Lightbend Inc.",
+    "Lightbend",
+    "com.lightbend.lagom",
+    "com.typesafe.play"
+  )
+
+  /**
+   * Comparable version information
+   */
+  final class Version(val version: String) extends Comparable[Version] {
+    private val (numbers: Array[Int], rest: String) = {
+      val numbers = new Array[Int](3)
+      val segments: Array[String] = version.split("[.-]")
+      var segmentPos = 0
+      var numbersPos = 0
+      while (numbersPos < 3) {
+        if (segmentPos < segments.length) try {
+          numbers(numbersPos) = segments(segmentPos).toInt
+          segmentPos += 1
+        } catch {
+          case e: NumberFormatException ⇒
+            // This means that we have a trailing part on the version string and
+            // less than 3 numbers, so we assume that this is a "newer" version
+            numbers(numbersPos) = Integer.MAX_VALUE
+        }
+        numbersPos += 1
+      }
+
+      val rest: String =
+        if (segmentPos >= segments.length) ""
+        else String.join("-", util.Arrays.asList(util.Arrays.copyOfRange(segments, segmentPos, segments.length): _*))
+
+      (numbers, rest)
+    }
+
+    override def compareTo(other: Version): Int = {
+      var diff = 0
+      diff = numbers(0) - other.numbers(0)
+      if (diff == 0) {
+        diff = numbers(1) - other.numbers(1)
+        if (diff == 0) {
+          diff = numbers(2) - other.numbers(2)
+          if (diff == 0) {
+            diff = rest.compareTo(other.rest)
+          }
+        }
+      }
+      diff
+    }
+
+    override def equals(o: Any): Boolean = o match {
+      case v: Version ⇒ compareTo(v) == 0
+      case _          ⇒ false
+    }
+
+    override def toString: String = version
+
+    override def hashCode(): Int = {
+      val state = Seq(numbers, rest, version)
+      state.map(_.hashCode()).foldLeft(0)((a, b) => 31 * a + b)
+    }
+  }
+}
+
+trait ProjectFamily {
+  def familyName: String
+  def libraries: immutable.Seq[String]
+}
+
+object AkkaHttpProjectFamily extends ProjectFamily {
+
+  val familyName = "Akka Http"
+
+  val libraries = List(
+    "akka-http",
+    "akka-http-bench-jmh",
+    "akka-http-caching",
+    "akka-http-core",
+    "akka-http-marshallers-java",
+    "akka-http-marshallers-scala",
+    "akka-http-testkit",
+    "akka-http-tests",
+    "akka-http2-support",
+    "akka-parsing"
+  )
+}
```

Comment:
Maybe place this information in META-INF/MANIFEST.MF and then we don't need to share any code between build tools and libraries.

https://www.scala-sbt.org/1.x/docs/Howto-Package.html#Add+manifest+attributes

Reply:
I think @marcospereira have suggested this at some point.

I don't remember why we realised this was less good, but we may be wrong. 

If the list added as attributes, we could scan the classloader and build the list of each family. That would also cover the case when a new module is added because the new module will offer a new entry under the same project family, which will be checked as well. 

But we still need to have a utility class able to scan the manifest and run the check, no?

Reply:
> But we still need to have a utility class able to scan the manifest and run the check, no?

Sure, but reusing that code across libraries and build tools isn't very important. Not much code, and not advanced, and will unlikely change.